### ISSUE TITLE
ggpairs has issue with variable called panel vs PANEL

### ISSUE THREAD

**Issue by sbihorel** [ggobi/ggally#289]
Hi,

ggpairs behaves incorrectly when a variable called 'PANEL' is part of the data. It is not the case when the variable is called 'panel' (lower case). Are there reserved variable names in GGally?

```
require(GGally)

gdata <- data.frame(
  age = rnorm(100,40,2),
  sexf = factor(sample(0:1, 100, replace = TRUE)),
  panel = factor(sample(1:3, 100, replace = TRUE)),
  study = factor(sample(1:3, 100, replace = TRUE))
)

ggpairs(gdata)

names(gdata) <- toupper(names(gdata))

ggpairs(gdata)
```

**Comment by schloerke** [comment]
Neat bug!  No. GGally does not have any reserved variable names.  

This is actually a ggplot2 issue...

``` r
library(ggplot2)

gdata <- data.frame(
  age = rnorm(100,40,2),
  sexf = factor(sample(0:1, 100, replace = TRUE)),
  panel = factor(sample(1:3, 100, replace = TRUE)),
  study = factor(sample(1:3, 100, replace = TRUE))
)

ggplot(gdata, aes(panel, age)) + geom_boxplot()
```

![](https://i.imgur.com/MGYn0d1.png)

``` r

gdata_upper <- gdata
names(gdata_upper) <- toupper(names(gdata_upper))

ggplot(gdata_upper, aes(PANEL, AGE)) + geom_boxplot()
```

![](https://i.imgur.com/xtkt943.png)

> Created on 2018-09-20 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by schloerke** [comment]
Original comment reprex:

``` r
library(GGally)
#> Loading required package: ggplot2
library(ggplot2)

gdata <- data.frame(
  age = rnorm(100,40,2),
  sexf = factor(sample(0:1, 100, replace = TRUE)),
  panel = factor(sample(1:3, 100, replace = TRUE)),
  study = factor(sample(1:3, 100, replace = TRUE))
)

ggpairs(gdata)
#> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
#> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
#> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
```

![](https://i.imgur.com/houZMVW.png)

``` r

gdata_upper <- gdata
names(gdata_upper) <- toupper(names(gdata_upper))

ggpairs(gdata_upper)
#> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
#> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
#> `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
```

![](https://i.imgur.com/WsJ0q6y.png)

> Created on 2018-09-20 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by schloerke** [comment]
closing as it is a ggplot2 error. Please follow at https://github.com/tidyverse/ggplot2/issues/2904

**Comment by schloerke** [comment]
tl;dr ggplot2 uses the variable PANEL. They recommend using a different name.
